graders/r/Dockerfile

```diff
@@ -1,19 +1,64 @@
-# Borrow the leg work done by the rocker project
-# c.f. https://github.com/rocker-org/rocker-versioned/tree/master/tidyverse
-FROM rocker/tidyverse:3.6.1
+# Image used for PrairieLearn external grading of R questions
+# as well as general support of STAT 477 Data Science Programming Methods
+# (which was formerly STAT 430 Topics - Data Science Programming Methods
 
-# Set a new label for the image
-LABEL org.label-schema.license="GPL-2.0" \
-      org.label-schema.vcs-url="https://github.com/PrairieLearn/PrairieLearn" \
-      org.label-schema.vendor="PrairieLearn" \
-      maintainer="James Joseph Balamuta <balamut2@illinois.edu>"
+# Alton Barbehenn and Dirk Eddelbuettel, 2019-2021
 
-# Add a script to install any additional R packages from CRAN or GitHub
-COPY r-requirements.R /
+# We base this container on the 'r-ubuntu:20.04' container of the
+# Rocker Project. This offers us a well-understood and stable basis
+# in the form of an Ubuntu LTS release, along with the a) ability to
+# deploy the current R version built on that release, and b) to source
+# several thousand CRAN packages as r-cran-* binaries via the PPA
+# See https://rocker-project.org for Rocker, and the README at
+# https://cran.r-project.org/bin/linux/ubuntu/ about R and the binaries
 
-# Modify the image to meet with PrairieLearn specifications
-# Enable autograde user for PrairieLearn and create home directory
-RUN adduser --disabled-password --gecos "" ag \                        
-    && su root -c "Rscript /r-requirements.R"
+FROM rocker/r-ubuntu:20.04
 
-# Rejoice! The docker image is built.
+# Needed to properly handle UTF-8
+ENV PYTHONIOENCODING=UTF-8
+
+# Install required libraries -- using prebuild binaries where available
+# We also install sqlite3 to support the SQL lectures
+RUN apt-get update && apt-get install -y \
+        curl \
+	git \
+	r-cran-data.table \
+	r-cran-devtools \
+        r-cran-diffobj \
+	r-cran-doparallel \
+	r-cran-dygraphs \
+	r-cran-foreach \
+	r-cran-fs \
+	r-cran-future.apply \
+	r-cran-gh \
+	r-cran-git2r \
+	r-cran-igraph \
+	r-cran-memoise \
+	r-cran-png \
+	r-cran-rcpparmadillo \
+	r-cran-rex \
+	r-cran-rsqlite \
+	r-cran-runit \
+	r-cran-shiny \
+	r-cran-stringdist \
+	r-cran-testthat \
+	r-cran-tidyverse \
+	r-cran-tinytest \
+	r-cran-xts \
+	sqlite3 \
+        sudo \
+        && echo "options(diffobj.brightness=\"dark\")" >> /etc/R/Rprofile.site
+
+# Install additional R packages from CRAN (on top of the ones pre-built as r-cran-*)
+RUN install.r bench flexdashboard microbenchmark lintr ttdo unix
+
+# Install plr -- for now (?) from GH; also install visualTest
+RUN installGithub.r stat447/plr MangoTheCat/visualTest
```

Comment:
Can we include `stat447/plr` directly into the `graders/r` directory? For example, like the way `graders/python/python_autograder` contains the python autograder library directly within `graders/python`. We prefer this because it keeps all the code for the grader together in the same repo.

Reply:
I guess I was hoping to have the dev version be the version directly in `graders/r/plr/...`.

But I'm also fine to keep it in a separate repo in the PL org. I can make `PrairieLearn/plr` if you'd rather do this? Let me know.

Reply:
Sorry -- I plainly misunderstood.

R packages have a very standard layout, and it (to my preferences, and to that of many others) works best when they sit top-level in their own repo.  `plr` is first and foremost an R package, which is why I am hesitant to have it a few levels down in the pl repo.  I would prefer not to.  (There are counter examples for other 'large scope multi-language projects' as _e.g._ Apache Arrow, or the MLPACK repo.  This is a little different.)

So yes `PrairieLearn/plr` (or another name) would be my preference.  It is currently an incomprehensibly short name ... because we had no prior constraints.  So the repo could be `PrairieLearn/pl-r-helpers` or `-misc` or ...

Reply:
Great! I created https://github.com/PrairieLearn/pl-r-helpers and you are Admin for it. I left the repo uninitialized in case you want to push the full history of `stat447/plr` into the new repo.

Let me know after you've switched the current PR over to use `PrairieLearn/pl-r-helpers` and I'll merge it.

Reply:
Awesome.  Was on a call for a bit.   We could also
- nuke pl-r-helpers (to create room)
- I transfer my existing plr
- we double check I can admin, else you do, and we rename it to pl-r-helpers
and that way keep the (trivial, but it is the right thing (tm)) history.  Worth it?

Reply:
If you have a checked-out local copy of `stat447/plr` then you should be able to do:
```
git remote rename origin old-origin
git remote add origin git@github.com:PrairieLearn/pl-r-helpers.git
git branch -M main
git push -u origin main
```

This will switch your checked-out repo to use `PrairieLearn/pl-r-helpers` as it's origin and then push a full copy of the history into this new repo.

Reply:
Doh. I was looking for something like that and I overcomplicated.  Decentralised revision control systems for the win, and a very nice wave of the clue bat!  I tried to do it all inside `magit` which fell apart when I realised I had not accepted your invitation so I couldn't actually write there.  Doh.  

All good now.

With that, back to installing the R package it represent  from git remote as before (now hitting a sibbling directory) ?

Reply:
Yep, jut switch to install from `PrairieLearn/pl-r-helpers`.

Reply:
Done in c371ea9de

Reply:
Thanks!